src/Magnum/Math/Geometry/Test/IntersectionTest.cpp

```diff
@@ -124,20 +144,192 @@ void IntersectionTest::pointFrustum() {
     CORRADE_VERIFY(!Intersection::pointFrustum({0.0f, 0.0f, 100.0f}, frustum));
 }
 
-void IntersectionTest::boxFrustum() {
+void IntersectionTest::rangeFrustum() {
     const Frustum frustum{
         {1.0f, 0.0f, 0.0f, 0.0f},
-        {-1.0f, 0.0f, 0.0f, 10.0f},
+        {-1.0f, 0.0f, 0.0f, 5.0f},
         {0.0f, 1.0f, 0.0f, 0.0f},
-        {0.0f, -1.0f, 0.0f, 10.0f},
+        {0.0f, -1.0f, 0.0f, 1.0f},
+        {0.0f, 0.0f, 1.0f, 0.0f},
+        {0.0f, 0.0f, -1.0f, 10.0f}};
+
+    /* Fully inside */
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{Vector3{1.0f}, Vector3{2.0f}}, frustum));
+    /* Intersects with exactly one plane each */
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{2.4f, -0.1f, 4.9f}, {2.6f, 0.1f, 5.1f}}, frustum));
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{2.4f, 0.9f, 4.9f}, {2.6f, 1.1f, 5.1f}}, frustum));
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{-0.1f, 0.4f, 4.9f}, {0.1f, 0.6f, 5.1f}}, frustum));
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{4.9f, 0.4f, 4.9f}, {5.1f, 0.6f, 5.1f}}, frustum));
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{2.4f, 0.4f, -0.1f}, {2.6f, 0.6f, 0.1f}}, frustum));
+    CORRADE_VERIFY(Intersection::rangeFrustum(Range3D{{2.4f, 0.4f, 9.9f}, {2.6f, 0.6f, 10.1f}}, frustum));
```

Comment:
What about using Range::fromSize() in the complex cases here? Would make it quite easier to maintain.

Reply:
Yeah, that would condense this a bit, I hope.